src/physics/convection_diffusion.cpp

```diff
@@ -169,44 +181,78 @@ template <int dim, int nstate, typename real>
 std::array<real,nstate> ConvectionDiffusion<dim,nstate,real>
 ::source_term (
     const dealii::Point<dim,real> &pos,
-    const std::array<real,nstate> &/*solution*/) const
+    const std::array<real,nstate> &/*solution*/,
+    const real current_time) const
 {
     std::array<real,nstate> source;
     const dealii::Tensor<1,dim,real> velocity_field = this->advection_speed();
     const real diff_coeff = diffusion_coefficient();
 
-    for (int istate=0; istate<nstate; istate++) {
-        dealii::Tensor<1,dim,real> manufactured_gradient = this->manufactured_solution_function->gradient (pos, istate);
-            // dealii::Tensor<1,dim,real> manufactured_gradient_fd = this->manufactured_solution_function.gradient_fd (pos, istate);
-            // std::cout<<"FD" <<std::endl;
-            // std::cout<<manufactured_gradient_fd <<std::endl;
-            // std::cout<<"AN" <<std::endl;
-            // std::cout<<manufactured_gradient <<std::endl;
-            // std::cout<<"DIFF" <<std::endl;
-            // std::cout<<manufactured_gradient - manufactured_gradient_fd <<std::endl;
-        dealii::SymmetricTensor<2,dim,real> manufactured_hessian = this->manufactured_solution_function->hessian (pos, istate);
-            // dealii::SymmetricTensor<2,dim,real> manufactured_hessian_fd = this->manufactured_solution_function.hessian_fd (pos, istate);
-            // std::cout<<"FD" <<std::endl;
-            // std::cout<<manufactured_hessian_fd <<std::endl;
-            // std::cout<<"AN" <<std::endl;
-            // std::cout<<manufactured_hessian <<std::endl;
-            // std::cout<<"DIFF" <<std::endl;
-            // std::cout<<manufactured_hessian - manufactured_hessian_fd <<std::endl;
-
-        //source[istate] = velocity_field*manufactured_gradient;
-        real grad = 0.0;
-        for (int d=0; d<dim; ++d) {
-            grad += velocity_field[d] * manufactured_gradient[d];
-        }
-        source[istate] = grad;
 
-        real hess = 0.0;
-        for (int dr=0; dr<dim; ++dr) {
-            for (int dc=0; dc<dim; ++dc) {
-                hess += (this->diffusion_tensor)[dr][dc] * manufactured_hessian[dr][dc];
+    using TestType = Parameters::AllParameters::TestType;
+
+    if(this->test_type == TestType::convection_diffusion_periodicity){
```

Comment:
similar comment to what I had left in `Burgers` about introducing a derived class

Reply:
fixed as per issue https://github.com/dougshidong/PHiLiP/issues/126